backend/src/org/akvo/lumen/endpoint/dashboard.clj

```diff
@@ -1,183 +1,33 @@
-             tx {:dashboard-id     dashboard-id
-                 :visualisation-id visualisation-id
-                 :layout           layout}))))
-      (handle-dashboard-by-id tenant-conn dashboard-id))
-    (throw (Exception. "Entities and layout dashboard keys does not match."))))
-
-
-
-;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
-;;; /dashboards/:id PUT
-;;;
-
-(defn persist-dashboard
-  "We update a dashboard via upsert of dashboard and clean - insert of
-  dashboard_visualisations.
-  1. Unpack text & visualisation entities
-  2. Update dashboard table
-  3. Remove old dashboard_visualisations
-  4. Add new dashboard_visualisations
-  "
-  [tenant-conn id spec]
-  (let [{texts          :texts
-         visualisations :visualisations} (part-by-entity-type spec)
-        visualisations-layouts           (:layout visualisations)]
-    (jdbc/with-db-transaction [tx tenant-conn]
-      (update-dashboard tx {:id    id
-                            :title (get spec "title")
-                            :spec  texts})
-      (delete-dashboard_visualisation tenant-conn {:dashboard-id id})
-      (doseq [visualisation-entity (:entities visualisations)]
-        (let [visualisation-id      (get visualisation-entity "id")
-              visualisations-layout (first (filter #(= visualisation-id
-                                                       (get % "i"))
-                                                   visualisations-layouts))]
-          (insert-dashboard_visualisation
-           tx {:dashboard-id     id
-               :visualisation-id visualisation-id
-               :layout           visualisations-layout}))))
-    (handle-dashboard-by-id tenant-conn id)))
-
-;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
-;;; /dashboards/:id DELETE
-;;;
-
-(defn handle-dashboard-delete [tenant-conn id]
-  (delete-dashboard_visualisation tenant-conn {:dashboard-id id})
-  (delete-dashboard-by-id tenant-conn {:id id})
-  {:status "OK"})
-
-
-;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
-;;; Routes
-;;;
 
 (defn endpoint [{:keys [tenant-manager]}]
 
   (context "/api/dashboards" {:keys [params tenant] :as request}
     (let-routes [tenant-conn (connection tenant-manager tenant)]
 
       (GET "/" _
-        (resp/response (all-dashboards tenant-conn)))
+        (response (dashboard/all tenant-conn)))
 
       (POST "/" {:keys [body]}
-        (resp/response (handle-new-dashboard tenant-conn body)))
+        (response (dashboard/create tenant-conn body)))
 
       (context "/:id" [id]
 
         (GET "/" _
-          (if-some [dashboard (dashboard-by-id tenant-conn {:id id})]
-            (resp/response (handle-dashboard-by-id tenant-conn  id))
-            (resp/not-found {:error "Not found"})))
+          (if-let [d (dashboard/fetch tenant-conn id)]
+            (response d)
+            (not-found {:error "not found"})))
```

Comment:
@kardan Is `not-found` an error? In some cases we return 404 + the `id` that wasn't found (for example in job execution). Not sure which is preferable? Perhaps the redundant error string `"not found"` is telling?